typegen/generate-types.js

```diff
@@ -0,0 +1,178 @@
+const { injectPropTypes, generateTypes } = require('prop-types-to-dts');
+const fs = require('fs');
+
+const PropTypes = injectPropTypes(require('prop-types'));
+
+const reactVis = require('../dist');
+
+//Adding missing prop types (described in docs but not covered with prop types of components)
+const colorType = PropTypes.oneOfType([PropTypes.string, PropTypes.number]);
+reactVis.AbstractSeries.propTypes = {
+  ...reactVis.AbstractSeries.propTypes,
+  color: colorType,
+  stroke: colorType,
+  fill: colorType,
+  opacity: PropTypes.number,
+}
+
+function splitPath(path) {
+  let [compName, ...propName] = path.split('.');
```

Comment:
typically the style of the library is to avoid lets as much as possible, either through variable renaming or just chains of stuff. I hadn't seen this usage of the ...operator before and I really like it, but just keep that let thing in mind